From 7f4f45efb44adffc8769cb2ef0d73a1f465da0bb Mon Sep 17 00:00:00 2001
From: Sean Paul <seanpaul@chromium.org>
Date: Tue, 30 Jan 2018 09:29:34 -0500
Subject: [PATCH 0998/2328] UPSTREAM: drm/i915: Restore HDCP DRM_INFO when with
 no downstream

The commit below returned earlier than before, but failed to move the
info message when authenticating without downstream devices. This patch
restores the message on authentication success.

Changes in v2:
- s/no downstream devices/no repeater present/ (Ram)

Fixes: 87eb3ec818fa ("drm/i915: II stage HDCP auth for repeater only")
Cc: Ramalingam C <ramalingam.c@intel.com>
Cc: Sean Paul <seanpaul@chromium.org>
Cc: Jani Nikula <jani.nikula@linux.intel.com>
Cc: Joonas Lahtinen <joonas.lahtinen@linux.intel.com>
Cc: Rodrigo Vivi <rodrigo.vivi@intel.com>
Cc: intel-gfx@lists.freedesktop.org
Reviewed-by: Daniel Vetter <daniel.vetter@ffwll.ch>
Signed-off-by: Sean Paul <seanpaul@chromium.org>
Link: https://patchwork.freedesktop.org/patch/msgid/20180130142943.168314-1-seanpaul@chromium.org
Link: https://patchwork.freedesktop.org/patch/msgid/20180130142943.168314-1-seanpaul@chromium.org
(cherry picked from commit 648a3b4ae0606efbceed3f726ac44e347aaef75e)

Change-Id: I1368704bbd0ce17c17c2e5e32c5ce2df0df6d77e
---
 drivers/gpu/drm/i915/intel_hdcp.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_hdcp.c b/drivers/gpu/drm/i915/intel_hdcp.c
index db95271..9fc9da0 100644
--- a/drivers/gpu/drm/i915/intel_hdcp.c
+++ b/drivers/gpu/drm/i915/intel_hdcp.c
@@ -168,10 +168,8 @@ int intel_hdcp_auth_downstream(struct intel_digital_port *intel_dig_port,
 
 	/* If there are no downstream devices, we're all done. */
 	num_downstream = DRM_HDCP_NUM_DOWNSTREAM(bstatus[0]);
-	if (num_downstream == 0) {
-		DRM_INFO("HDCP is enabled (no downstream devices)\n");
+	if (num_downstream == 0)
 		return 0;
-	}
 
 	ksv_fifo = kzalloc(num_downstream * DRM_HDCP_KSV_LEN, GFP_KERNEL);
 	if (!ksv_fifo)
@@ -502,6 +500,7 @@ static int intel_hdcp_auth(struct intel_digital_port *intel_dig_port,
 	if (repeater_present)
 		return intel_hdcp_auth_downstream(intel_dig_port, shim);
 
+	DRM_INFO("HDCP is enabled (no repeater present)\n");
 	return 0;
 }
 
-- 
2.7.4

